test/strategies/cloudsearch.js

```diff
@@ -0,0 +1,151 @@
+'use strict';
+
+// Load external modules
+const Lab = require('lab');
+const MockAWS = require('mock-aws');
+
+// Load internal modules
+const CloudSearchStrategy = require('../../src/strategies/cloudsearch');
+
+// Test shortcuts
+const lab = exports.lab = Lab.script();
+const expect = Lab.assertions.expect;
+
+let strategy;
+
+lab.after((done) => {
+  MockAWS.restore();
+  done();
+});
+
+lab.before((done) => {
+  strategy = new CloudSearchStrategy({
+    region: 'us-west-1',
+    endpoint: 'https://www.ideapod.com'
+  });
+  done();
+});
+
+lab.describe('CloudSearchStrategy', () => {
+  lab.describe('uploadDocuments', () => {
+    lab.beforeEach((done) => {
+      MockAWS.mock('CloudSearchDomain', 'uploadDocuments', (params, call) => {
+        if (!params) {
+          return call(new Error('no params'));
+        }
+        return call(null, { status: true });
+      });
+      done();
+    });
+    lab.it('upload a document', (done) => {
+      const params = {
+        contentType: 'application/json',
+        documents: JSON.stringify([{ 'bar': 'foo' }])
+      };
+      strategy.uploadDocuments(params)
+        .then((data) => {
+          expect(data.status).to.be.true();
+          done();
+        })
+        .catch(done);
+    });
+    lab.it('reject error if no params', (done) => {
+      strategy.uploadDocuments()
+        .catch((err) => {
+          expect(err).to.be.object();
+          done();
+        });
+    });
+  });
+
+  lab.describe('search', () => {
+    lab.beforeEach((done) => {
+      MockAWS.mock('CloudSearchDomain', 'search', (params, call) => {
+        if (!params) {
+          return call(new Error('no params'));
+        }
+        return call(null, { status: true });
+      });
+      done();
+    });
+    lab.it('search in the cloud', (done) => {
+      strategy.search({ title: 'test' })
+        .then((data) => {
+          expect(data).to.be.object();
+          done();
+        })
+        .catch(done);
+    });
+    lab.it('requires params', (done) => {
+      strategy.search()
+        .catch((err) => {
+          expect(err).to.be.object();
+          done();
+        });
+    });
+  });
+
+  lab.describe('addDocument', () => {
+    let testIdea;
```

Comment:
Remove these. It's very hard to follow what's happening in the tests if you keep assigning data to outside variables.